File: credentials-verification-android/app/src/main/java/io/iohk/cvp/views/utils/dialogs/SuccessDialog.java
package io.iohk.cvp.views.utils.dialogs;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;

import java.util.Objects;

import io.iohk.cvp.R;

public class SuccessDialog extends DialogFragment {

  private Fragment context;

  public static SuccessDialog newInstance(Fragment context, int description) {
    SuccessDialog frag = new SuccessDialog();
    Bundle args = new Bundle();
    args.putInt("description", description);
    frag.context = context;
    frag.setArguments(args);
    return frag;
  }


  @Override
  public Dialog onCreateDialog(Bundle savedInstanceState) {
    int description = getArguments().getInt("description");

    // Use the Builder class for convenient dialog construction
    AlertDialog.Builder builder = new AlertDialog.Builder(Objects.requireNonNull(getActivity()));

    LayoutInflater inflater = context.getLayoutInflater();
    View dialogView = inflater.inflate(R.layout.success_popup, null);

    TextView descriptionTextView = dialogView.findViewById(R.id.description_text_view);
    descriptionTextView.setText(description);

    Button btnOk = dialogView.findViewById(R.id.btn_ok);
    btnOk.setOnClickListener(new View.OnClickListener() {
      @Override
      public void onClick(View view) {
        SuccessDialog.this.dismiss();
      }
    });

    builder.setView(dialogView);
    return builder.create();
  }
}

File: prism-android-wallet/app/src/main/java/io/iohk/atala/prism/app/ui/commondialogs/SuccessDialog.kt
package io.iohk.atala.prism.app.ui.commondialogs

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.DialogFragment
import io.iohk.cvp.R
import io.iohk.cvp.databinding.DialogSuccessBinding

class SuccessDialog : DialogFragment() {

    class Builder(private val context: Context) {
        private val dialog = SuccessDialog()

        fun setPrimaryText(resId: Int): Builder = setPrimaryText(context.getString(resId))

        fun setPrimaryText(text: String): Builder {
            dialog.primaryText = text
            return this
        }

        fun setSecondaryText(resId: Int): Builder = setSecondaryText(context.getString(resId))

        fun setSecondaryText(text: String): Builder {
            dialog.secondaryText = text
            return this
        }

        fun setCustomButtonText(resId: Int): Builder = setCustomButtonText(context.getString(resId))

        fun setCustomButtonText(text: String): Builder {
            dialog.customButtonText = text
            return this
        }

        fun setOkButtonClickLister(listener: View.OnClickListener): Builder {
            dialog.okButtonClickListener = listener
            return this
        }

        fun build(): SuccessDialog = dialog
    }

    private lateinit var binding: DialogSuccessBinding

    private var primaryText: String? = null

    private var secondaryText: String? = null

    private var customButtonText: String? = null

    private var okButtonClickListener: View.OnClickListener? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setStyle(STYLE_NO_TITLE, R.style.AlertDialogTheme) // add App custom style
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        isCancelable = false
        binding = DialogSuccessBinding.inflate(inflater, container, false)
        binding.lifecycleOwner = this
        binding.primaryText = primaryText
        binding.secondaryText = secondaryText
        binding.customButtonText = customButtonText
        binding.okButton.setOnClickListener {
            dismiss()
            okButtonClickListener?.onClick(it)
        }
        dialog?.window?.setBackgroundDrawableResource(R.drawable.rounded_corner_white)
        return binding.root
    }
}
